Sources/SourceKitD/SourceKitD.swift

```diff
@@ -83,26 +83,22 @@ extension SourceKitD {
   public func send(_ req: SKDRequestDictionary) async throws -> SKDResponseDictionary {
     logRequest(req)
 
-    return try await withCheckedThrowingContinuation { continuation in
+    let sourcekitdResponse: SKDResponse = await withCheckedContinuation { continuation in
       var handle: sourcekitd_request_handle_t? = nil
 
-      api.send_request(req.dict, &handle) { [weak self] _resp in
-        guard let self = self else { return }
-
-        let resp = SKDResponse(_resp, sourcekitd: self)
-
-        logResponse(resp)
-
-        guard let dict = resp.value else {
-          continuation.resume(throwing: resp.error!)
-          return
-        }
-
-        continuation.resume(returning: dict)
+      api.send_request(req.dict, &handle) { _resp in
+        continuation.resume(returning: SKDResponse(_resp, sourcekitd: self))
```

Comment:
Is dropping the weak-strong dance intentional here?

Reply:
Actually I suppose the old implementation was wrong because it could have lead to the task being suspended indefinitely

Reply:
Yeah, I looked at it and decided that it isn’t needed to prevent reference cycles. 

Good catch with `ClangLanguageServerShim.shutdown`, I’ll clean that up as well.